### PR TITLE
Feature 21/coop registration

### LINKED ISSUE #21: User sign up
As a Chairman of Сo-owners of apartment building association I want to register cooperation in the system.

**Preconditions:**
1. Сo-owners of apartment building association shall have National State Registry of Ukrainian Enterprises and Organizations code.
2. Chairman of Сo-owners of apartment building association has to be chosen.

 **Acceptance criteria**

1. Button “Register cooperation” redirect to the window with the fields requires to input e-mail and National State Registry of Ukrainian Enterprises and Organizations code.
2. When e-mail or National State Registry of Ukrainian Enterprises and Organizations code is already registered in the system - the system shall propose user login.
3. When e-mail and National State Registry of Ukrainian Enterprises and Organizations code input correctly button “Register” becomes available.
4. By pressing the “Register” button confirmation e-mail shall be sent. 

**Fields fulfill requirements**
Email address field: Max email length – 320 symbols, Min email length - 5 symbols, Recipient name shall consist of uppercase and lowercase letters in English (A-Z, a-z), digits from 0 to 9, special characters ! # $ % & ' * + - / = ? ^ _ ` { | ( ) , : ; < > @ [ \ ]
A special character cannot appear as the first or last character in an email address or appear consecutively two or more times.
Presence of one @ symbol is obligatory
Domain name shall consist of uppercase and lowercase letters in English (A-Z, a-z), digits from 0 to 9, hyphen (-), a period (.)
The address has to have a period (.) symbol placed after the domain name before the top top-level domain in an email address.

Organizations code field: length - 8 symbols, digits from 0 to 9.

[https://home-project-engineering.atlassian.net/browse/HP-12](url)

Coop_register

### PR DESCRIPTION
**Pull request by tsehla42**
## Summary of change

added popups
moved header usage to app component
enhanced validation
error handling on failing requests

### CODE REVIEW

**Review by markuwas — DISMISSED** [review]

**Comment by what-the-hub** [comment]
Hello! Your code looks good for me, We have some bug with header (BaseHeader both in router and app) in home page but i think its not your task

File: src/App.vue [hunk 1]
```diff
@@ -1,18 +1,18 @@
 <template>
+  <BaseHeader />
   <router-view></router-view>
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
+import BaseHeader from '@/components/base/BaseHeader.vue';
```
Comment (what-the-hub) [inline src/App.vue]: i think we already got this component in router

**Review by what-the-hub — COMMENTED** [review]
just saw that you add baseheader in app. Check the homepage

File: src/App.vue (hunk 1 above)
Comment (tsehla42) [reply src/App.vue]: As I said on daily meeting 20.09, we should move header to app component and delete it from other places.

File: src/views/RegisterCooperation.vue [hunk 2]
```diff
@@ -71,25 +90,81 @@ export default defineComponent({
...
+        this.check.edrpou.valid = false;
+        if (isValid) {
+          this.check.edrpou.valid = true;
+          this.checkEdrpouRegistered();
+        }
+      });
+    },
+    checkEmailRegistered() {
+      this.$http
+        .get('/usersы', { params: { email: this.email } })
```
Comment (maveric-ui) [inline src/views/RegisterCooperation.vue]: first of all should be in vuxe store
seems path with a mistake `usersы` and will not work

File: src/views/RegisterCooperation.vue [hunk 3]
```diff
@@ -71,25 +90,81 @@ export default defineComponent({
...
+        .then((r) => {
+          this.check.email.registered = r.data.length !== 0;
```
Comment (maveric-ui) [inline src/views/RegisterCooperation.vue:134]: it totally suitable for getters in vuex store

File: src/views/RegisterCooperation.vue [hunk 4]
```diff
@@ -71,25 +90,81 @@ export default defineComponent({
...
+        })
+        .catch(this.showError);
+    },
+    checkEdrpouRegistered() {
+      this.$http
+        .get('/cooperation', { params: { usreo: this.edrpou } })
+        .then((r) => {
```
Comment (maveric-ui) [inline src/views/RegisterCooperation.vue]: what a type of response

File: src/views/RegisterCooperation.vue [hunk 5]
```diff
@@ -71,25 +90,81 @@ export default defineComponent({
...
+          this.check.edrpou.registered = r.data.length !== 0;
```
Comment (maveric-ui) [inline src/views/RegisterCooperation.vue:147]: same, should be in vuex getters

**Review by markuwas — DISMISSED** [review]

**Review by what-the-hub — APPROVED** [review]

**Review by lexysline — APPROVED** [review]

**Pull request merged**